File: languages/JS/prime.js
function isPrime(n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 === 0 || n % 3 === 0) return false;
    let i = 5;
    while (i * i <= n) {
        if (n % i === 0 || n % (i + 2) === 0) return false;
        i += 6;
    }
    return true;
}

const startTime = Date.now();
for (let i = 2; i < 10000000; i++) {
    if (isPrime(i)) {
        console.log(i);
    }
}
const endTime = Date.now();
console.log((endTime - startTime) / 1000, "seconds to execute");

File: languages/Nim/prime.nim
import times

proc isPrime(n: int): bool =
    if n <= 1:
        return false
    if n <= 3:
        return true
    if n mod 2 == 0 or n mod 3 == 0:
        return false
    var i = 5
    while i * i <= n:
        if n mod i == 0 or n mod (i + 2) == 0:
            return false
        i += 6
    return true

let startTime = cpuTime()
for i in 2 ..< 10000000:
    if isPrime(i):
        echo i
let endTime = cpuTime()
echo (endTime - startTime), " seconds to execute"
